### PR TITLE
ENH: Adds security_start_date and _end_date with deprecation warnings

### PR DESCRIPTION
**Pull request by jfkirk**
These fields are required for compatibility with existing Quantopian algorithms. They are marked for deprecation in the future if possible.

### CODE REVIEW

**Comment by jfkirk** [comment]
@richafrank Could you review this?

**Comment by richafrank** [comment]
Sure thing.  We should get @ssanderson 's input too.

File: zipline/assets/_assets.pyx [hunk 1]
```diff
@@ -203,6 +203,26 @@ cdef class Equity(Asset):
         params = ', '.join(strings)
         return 'Equity(%d, %s)' % (self.sid, params)
 
+    property security_start_date:
+        """
+        DEPRECATION: This property should be deprecated and is only present for
```
Comment (richafrank) [inline zipline/assets/_assets.pyx]: Are we intentionally choosing not to deprecate it now, or can we warn with a DeprecationWarning at runtime instead?  Or maybe this is a cython convention? (I don't know too much cython.)

Comment (jfkirk) [inline zipline/assets/_assets.pyx]: I'm unfamiliar with deprecation in general - is there some way that we could make calls to security_start_date throw deprecation warnings in to the user log?

Comment (richafrank) [inline zipline/assets/_assets.pyx]: Yea, you can use [warnings.warn] (https://docs.python.org/2/library/warnings.html#warnings.warn), passing DeprecationWarning.  I think we have examples of this elsewhere in the code.

Comment (jfkirk) [inline zipline/assets/_assets.pyx]: Cool, thank you. I've added a warning below to both properties.

Comment (richafrank) [inline zipline/assets/_assets.pyx]: Awesome!  What do you think of DeprecationWarning instead of UserWarning?

Comment (jfkirk) [inline zipline/assets/_assets.pyx]: Thanks - wasn't aware of that class. Done.

**Comment by richafrank** [comment]
lgtm @jfkirk !  Sound good to you, @ssanderson ?

**Comment by ssanderson** [comment]
I'm not sure I understand why we have "asset/security_start_date" as distinguished from just start_date and end_date...changes like this seem like a strong argument for just calling the field "start_date" or "end_date" so that they don't become retroactively incorrect if we change names.

Is there code that currently depends on the use of `asset_start_date` as opposed to just `start_date`?

**Comment by richafrank** [comment]
Assuming we keep security_start_date for now, should we add a unit test for it?

**Comment by jfkirk** [comment]
I can add a unit test.

@ssanderson I can cut asset_start_date to just start_date, but we will still need to provide this near-deprecated security_start_date so as to not break old algos.

**Comment by jfkirk** [comment]
@richafrank Added a test
@ssanderson I've removed the asset_*_date fields in deference to simply start_date and end_date. I'm checking impact on qexec now.

**Comment by richafrank** [comment]
Awesome - thanks!

**Pull request merged**